src/coreclr/jit/gentree.h

```diff
@@ -3652,6 +3652,8 @@ struct GenTreeLclVarCommon : public GenTreeUnOp
         return m_ssaNum.IsComposite();
     }
 
+    bool IsNeverNegative(Compiler* comp) const;
```

Comment:
This would be more appropriate on `GenTreeLclVar` given what it does.

Reply:
It's something available on `LclVarDsc` and which could be extended to any other type in the future.

We have many functions which are the lowest common type where they make sense or where they are convenient to expose.

Reply:
Fixed based on the conversation above.